### PR TITLE
Add support to Zipkin V1

### LINKED ISSUE #250: receiver/zipkin: Add support for v1 API
Based on #138

### PR DESCRIPTION
**Pull request by pjanotti**
Fixes #250 

Add a trace translator from Zipkin V1 to OC proto.
Per ealier dicussion we just need to support JSON,
this gets OC service working with current Envoy tracing
for Zipkin.

### CODE REVIEW

**Comment by songy23** [comment]
This PR is too large - maybe create a separate PR to check in the test data first?

**Comment by pjanotti** [comment]
@songy23 The test data came straight from Envoy but there was a bit of redundant stuff in terms of code coverage. Removed redundant parts keeping the core annotations to document the type of annotation that we expect to get. That is about 550 less lines of code on the tests.

File: receiver/zipkin/proto_parse_test.go [hunk 1]
```diff
@@ -102,9 +102,10 @@ func TestConvertSpansToTraceSpans_protobuf(t *testing.T) {
 	zi := new(ZipkinReceiver)
 	hdr := make(http.Header)
 	hdr.Set("Content-Type", "application/x-protobuf")
+	r := &http.Request{Header: hdr}
```
Comment (odeke-em) [inline receiver/zipkin/proto_parse_test.go]: If we keep the multiplexing in ServeHTTP while cleanly separating the methods for v1 and v2, then we don't have to manually create this Request or make this line change and make a specific unit test for v1.

If we want to ensure that they properly multiplex, we can just use ServeHTTP in tests like this
```go
req := http.NewRequest("POST", "http://opencensus.io/", blob)
req.Header.Set("Content-Type", CONTENT_TYPE)
rec := httptest.Recorder()
zi.ServeHTTP(rec, req)
res := rec.Result()
zBlob, _ := ioutil.ReadAll(res.Body)
zreqs, err := json.Unmarshal(zBlob, &recv)
```

File: translator/trace/zipkinv1_to_protospan.go [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+	// Unknow service name works both as a default value and a flag to indicate that a valid endpoint was found.
+	const unknownServiceName = "unknown-service"
+
+	// Zipkin V1 annotations have a timestamp so they fit well with OC TimeEvent
+	earlyAnnotationTimestamp := int64(math.MaxInt64)
+	lateAnnotationTimestamp := int64(math.MinInt64)
+	res := &annotationParseResult{}
+	timeEvents := make([]*tracepb.Span_TimeEvent, 0, len(annotations))
+	for _, currAnnotation := range annotations {
+		if currAnnotation == nil && currAnnotation.Value != "" {
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply translator/trace/zipkinv1_to_protospan.go]: Good catch!

File: receiver/zipkin/trace_receiver.go [hunk 3]
```diff
@@ -110,13 +112,30 @@ func (zr *ZipkinReceiver) StartTraceReception(ctx context.Context, spanSink rece
 	return err
 }
 
-func (zr *ZipkinReceiver) parseAndConvertToTraceSpans(blob []byte, hdr http.Header) (reqs []*agenttracepb.ExportTraceServiceRequest, err error) {
+func (zr *ZipkinReceiver) parseAndConvertToTraceSpans(blob []byte, r *http.Request) (reqs []*agenttracepb.ExportTraceServiceRequest, err error) {
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply receiver/zipkin/trace_receiver.go]: Yeah, it will be better with two separate functions.

**Comment by pjanotti** [comment]
Thanks for the review @odeke-em, code looks better now 👍

**Pull request merged**